### ISSUE TITLE
Broken handling of conditional expressions

### ISSUE THREAD

**Issue by paralogismos** [tkurtbond/old-ada-mode#3]
It seems that old-ada-mode is broken when handling `case` expressions and `if` expressions. Consider the two code examples below:

```ada
with Ada.Text_IO; use Ada.Text_IO;

procedure Case_Expr is
   type Pet is (Dog, Cat, Goldfish);
   My_Pet : Pet := Dog;
begin
   Put_Line(
     (case My_Pet is
        when Dog => "Doggie",
        when Cat => "Kitty",
        when Goldfish => "Fishy"));
end Case_Expr;
```

```ada
with Ada.Text_IO; use Ada.Text_IO;

procedure If_Expr is
   type Pet is (Dog, Cat, Goldfish);
   My_Pet : Pet := Dog;
begin
   Put_Line(
     (if My_Pet = Dog then "Doggie"
     elsif My_Pet = Cat then "Kitty"
     else "Goldfish"));
end If_Expr;
```

These are legal Ada programs which compile and run, but old-ada-mode complains about _**"No matching proc/func/task/declare/package/protected."**_ When the conditional expression occurs in a loop old-ada-mode complains of the `end loop` that _**"Matching start is not 'loop'."**_

Note that old-ada-mode can be satisfied by the inclusion of a "matching" `end case` or `end if`, but then the code is no longer legal Ada code and will not compile.

This issue does not prevent one from writing correct code, but it does cause much unpleasantness and havoc with formatting and ada-mode warnings below any occurrences of conditional expressions in source code.

Conditional expressions were an addition to Ada 2012, so I suspect that old-ada-mode just predates these newer language features; still, this is an annoying gap.

**Comment by mgrojo** [comment]
Yes, the old `ada-mode` did not support Ada 2012, that should be stated in the README.

We have three possible Ada modes for Emacs, and none of them entirely satisfactory, in my opinion. The third one is  I've checked that [`ada-light-mode`](https://github.com/sebastianpoeplau/ada-light-mode) indents your snippets (although rigidly and after finishing the current statement) once you've activated `eglot` and the `ada_language_server`.
